modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -147,6 +161,12 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
         ceilMode = params.get<bool>("ceil_mode", true);
         spatialScale = params.get<float>("spatial_scale", 1);
         avePoolPaddedArea = params.get<bool>("ave_pool_padded_area", true);
+
+        if(type == L1 || type == L2 || type == PROD || type == LOG_SUM_EXP || type == LOG_SUM || type == SUM_SQUARE )
+            for(auto & kernel_size_i : kernel_size)
+            {
+                CV_Assert(kernel_size_i == 1);
+            }
```

Comment:
Hi, @rogday. Can you check if these codes are written in the right way?

Reply:
Yes, I think so. I was concerned because later in the forward we sometimes divide temporary value by the are of kernel and I think it doesn't make sense to do so in these cases.